### PR TITLE
Finished Feature/dirac.x

### LINKED ISSUE #35: Implement dirac.x application
Using the basic "identity" `umdsstCovar` implemented by #34, get the `dirac.x` application working.

- check `Increment::dirac()` to make sure it working correctly. It should use the x,y coordinates in the yaml configuration to set that grid box of the increment field to 1.0. Since we have an MPI decomposed grid, you'll need to convert the global x,y to a local x,y and only do something with it if it is indeed on the PE's local section of the grid. (i'm sure atlas provides the infromation somewhere for converting between global/local indexes)
- enable the ctest for dirac.x

at this point dirac test should produce output files that show increments at the exact x,y locations specified in the yaml. The increment will not yet be spread out horizontally.

### PR DESCRIPTION
**Pull request by loganchen39**
@travissluka Passed all tests. Some points below.
1. I set 8 grid points which are across the domain for testing purpose, see dirac.yml.
2. As I checked, there's no easy and direct mapping from our specification (e.g. (i=20, j=10)) to 1D local_index, we need to first compute its 1D global_index and then find the corresponding local_index by search. We could use more efficient search algorithm (e.g. binary search) in the future if certain conditions are met for sure, e.g. the array is increasing. For now I used the simplest linear search algo. 
3. The output netCDF file consists of two values, 273.15 and 274.15, which corresponds to 0 and 1. Bear it for now. The Fields::write() function converts from Celsius to Kelvin, as when we were implementing class State, its input netCDF file is in Kelvin, so the Fields::write() function has to output Kelvin as well, or it won't pass tests for class State. We may need to change it in the future, e.g. convert all of our SST netCDF files from Kelvin to Celsius.

close #35 .

### CODE REVIEW

File: test/testref/dirac_1GridPoint.ref [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Test     : Input Dirac increment: min = 0, max = 1, mean = 3.08642e-05
```
Comment (travissluka) [inline test/testref/dirac_1GridPoint.ref]: is this file being used for anything? if not, delete

File: test/testinput/dirac.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+background error:
+  covariance model: umdsstCovar
+  horizontal_length_scale: 2.2e6
+  maximum_condition_number: 1.0e6
+  standard_deviation: 1.8e7
+dirac:
+  date: 2010-01-01T12:00:00Z
+  ixdir: [20, 50, 100, 150, 200, 250, 300, 350]
+  iydir: [10, 25, 50,  75,  100, 125, 150, 175]
+geometry:
```
Comment (travissluka) [inline test/testinput/dirac.yml:10]: `dirac.yml` needs to be added to the relevant section of the `test/CMakeLists.txt` file https://github.com/UMD-AOSC/UMD-SST/blob/40244f0b0a95e66bd086b9269fe980e161f30cc5/test/CMakeLists.txt#L1

It is not being linked into the build/test directory and so the ctests are currently failing

**Review by travissluka — CHANGES_REQUESTED** [review]

File: test/testinput/dirac.yml (hunk 2 above)
Comment (travissluka) [reply test/testinput/dirac.yml:10]: ... same thing with the `testref/dirac.ref` file

**Comment by travissluka** [comment]
@loganchen39 otherwise, looks good :+1:, and i'll merge once the above issues are addressed so that the ctests pass

**Comment by loganchen39** [comment]
@travissluka I've fixed these minor problems. But it seems the Cheyenne environment has been changed and I can not run tests or use some git commands (e.g. "git push"). Will let you know once it's solved. Merry Christmas and Happy New Year!

File: test/testref/dirac_1GridPoint.ref (hunk 1 above)
Comment (loganchen39) [reply test/testref/dirac_1GridPoint.ref]: It's deleted.

File: test/testinput/dirac.yml (hunk 2 above)
Comment (loganchen39) [reply test/testinput/dirac.yml:10]: Done.

**Comment by loganchen39** [comment]
@travissluka Mark just fixed the Cheyenne problem. I re-compiled and re-run the tests, it seems ok. Please try to approve the PR. Thanks,

**Review by travissluka — APPROVED** [review]
TravisCI tests should pass once https://github.com/UMD-AOSC/UMD-SST/pull/45 is merged in first

**Pull request merged**